cmd/dep/gopath_scanner.go

```diff
@@ -53,8 +53,9 @@ func (g *gopathScanner) InitializeRootManifestAndLock(rootM *dep.Manifest, rootL
 	}
 
 	g.origM = &dep.Manifest{
-		Constraints: g.pd.constraints,
-		Ovr:         make(gps.ProjectConstraints),
+		Constraints:  g.pd.constraints,
+		Ovr:          make(gps.ProjectConstraints),
+		PruneOptions: gps.PruneNestedVendorDirs,
```

Comment:
I've already added a Manifest constructor based on your comments, seems like a great idea. 
I've updated the importers to use the new constructor. :grin:

I'll rebase and restructure the commits to allow sane review since this PR contains a lot of changes. 😓